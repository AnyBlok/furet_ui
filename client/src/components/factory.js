import Vue from 'vue';

const components = {};
const componentNames = [];

export const defineComponent = (elementName, declaration) => {
  let klass;
  if (elementName in components) {
    klass = components[elementName];
  } else {
    klass = {
      template: undefined,
      extend: [],
      prototypes: [],
    };
    components[elementName] = klass;
    componentNames.push(elementName);
  }
  if (declaration.template !== undefined) klass.template = declaration.template;
  if (declaration.prototype !== undefined) {
    klass.prototypes.push(declaration.prototype);
  }
  if (declaration.extend !== undefined) {
    declaration.extend.forEach((extend) => {
      klass.extend.push(extend);
    });
  }
};

window.defineComponent = defineComponent;

const getPrototypeFor = (elementName) => {
  const component = components[elementName];
  const prototypes = [];
  component.extend.forEach((otherElementName) => {
    prototypes.push(...getPrototypeFor(otherElementName));
  });
  prototypes.push(...component.prototypes);
  return prototypes;
};

export const createComponents = () => {
  componentNames.forEach((elementName) => {
    const component = components[elementName];
    const mixins = getPrototypeFor(elementName);
    Vue.component(elementName, (resolve) => {
      resolve({
        template: component.template,
        mixins,
      });
    });
  });
};
