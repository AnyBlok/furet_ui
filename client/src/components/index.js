import './mixins';
import './app';
import './homepage';
import './login';
import './logout';
import './about';
import './list';

export { createComponents, default } from './factory';
