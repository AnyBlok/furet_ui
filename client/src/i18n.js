import Vue from 'vue';
import VueI18n from 'vue-i18n';

Vue.use(VueI18n);

export const i18nConf = {
  locale: 'en',
  fallbackLocale: 'en',
  messages: {
    en: {
    },
  },
};
export const i18n = new VueI18n(i18nConf);
