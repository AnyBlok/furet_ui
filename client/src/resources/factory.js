import _ from 'underscore';

const resources = {};
export const resourceNames = [];

export const removeResource = (resourceName) => {
  const index = resourceNames.indexOf(resourceName);
  if (index !== -1) {
    delete resources[resourceName];
    resourceNames.splice(index, 1);
  }
};

export const defineResource = (resourceName, declaration) => {
  let klass;
  if (resourceName in resources) {
    klass = resources[resourceName];
  } else {
    klass = {
      children: [],
      name: resourceName,
      path: resourceName,
      templateName: resourceName,
      props: [],
      mustBeAuthenticated: false,
    };
    resources[resourceName] = klass;
    resourceNames.push(resourceName);
  }
  if (declaration.path !== undefined) klass.path = declaration.path;
  if (declaration.templateName !== undefined) klass.templateName = declaration.templateName;
  if (declaration.props !== undefined) klass.props = declaration.props;
  if (declaration.mustBeAuthenticated !== undefined) {
    klass.mustBeAuthenticated = declaration.mustBeAuthenticated;
  }
  if (declaration.children !== undefined) {
    declaration.children.forEach((child) => {
      let childDefinition = _.find(klass.children, c => c.name === child.name);
      if (childDefinition === undefined) {
        childDefinition = {
          name: child.name,
          completeName: `${resourceName}_${child.name}`,
          templateName: child.name,
          path: child.name,
          props: [],
          mustBeAuthenticated: false,
        };
        klass.children.push(childDefinition);
      }
      if (child.path !== undefined) childDefinition.path = child.path;
      if (child.templateName !== undefined) childDefinition.templateName = child.templateName;
      if (child.props !== undefined) childDefinition.props.push(...child.props);
      if (child.mustBeAuthenticated !== undefined) {
        childDefinition.mustBeAuthenticated = child.mustBeAuthenticated;
      }
    });
  }
};

window.defineResource = defineResource;

const createRoute = (resource, defaultTemplateProps) => {
  let path = `/${resource.path}`;
  let templateProps = defaultTemplateProps || '';
  if (resource.path === resource.name) {
    if (resource.props.length !== 0) {
      resource.props.forEach((prop) => {
        path += `/:${prop}`;
        templateProps += ` v-bind:${prop}="${prop}"`;
      });
    }
  } else if (resource.props.length !== 0) {
    resource.props.forEach((prop) => {
      templateProps += ` v-bind:${prop}="${prop}"`;
    });
  }
  const route = {
    name: resource.completeName || resource.name,
    path,
    props: templateProps !== '',
    meta: { requiresAuth: resource.mustBeAuthenticated },
    component: { props: resource.props },
  };
  if (resource.children === undefined || resource.children.length === 0) {
    route.component.template = `<${resource.templateName}${templateProps} />`;
  } else {
    route.children = [];
    if (resource.templateName === resource.name) {
      route.component.template = '<router-view />';
    } else {
      route.component.template = `<${resource.templateName}${templateProps} />`;
    }
    resource.children.forEach((child) => {
      route.children.push(createRoute(child, templateProps));
    });
  }
  return route;
};

export const getRoutes = () => {
  const routes = [];
  resourceNames.forEach((resourceName) => {
    const resource = resources[resourceName];
    routes.push(createRoute(resource));
  });
  return routes;
};
