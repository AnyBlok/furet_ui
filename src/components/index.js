import './mixins';
import './app';
import './homepage';
import './login';
import './logout';
import './pages';
import './fields';

export { createComponents, default } from './factory';
