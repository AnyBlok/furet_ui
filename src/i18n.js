/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import Vue from 'vue';
import VueI18n from 'vue-i18n';
Vue.use(VueI18n);


export const i18nConf = {
    locale: 'en',
    fallbackLocale: 'en',
    messages: {
        en: {
            menus: {
                close: 'Close',
                search: 'Search ...',
            },
            views: {
                unknown: {
                    title: 'The wanted view "{name}" is unknown',
                    message: "Report this message to the administrator",
                },
                common: {
                    create: 'Create',
                    save: 'Save',
                    edit: 'Edit',
                    cancel: 'Cancel',
                    delete: 'Delete',
                    close: 'Close',
                    actions: 'Actions',
                    more: 'More',
                },
                clients: {
                    login: {
                        button: 'Log in',
                    },
                },
            },
            fields: {
                common: {
                    required: 'this field is required',
                },
                json: {
                    invalid: 'Invalid JSON format',
                },
                file: {
                    upload: 'Upload',
                    download: 'Download',
                    delete: 'Delete',
                },
            },
        },
    },
};
export const i18n = new VueI18n(i18nConf);
