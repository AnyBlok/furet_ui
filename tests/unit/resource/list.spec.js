import { mount } from "@vue/test-utils";
import { getComponentPrototype } from '@/components/factory';
import { i18n } from "@/i18n";
// import { eval_counter } from "@/components/resource/helper";
import axios from 'axios';

const localVue = global.localVue;
const router = global.router;

const mocki18n = jest.spyOn(i18n, "_t");
mocki18n.mockResolvedValue('');

const mockAxios = jest.spyOn(axios, "post");
mockAxios.mockImplementationOnce(() =>
  Promise.resolve({ data: "redirect/uri" })
);
const mockAxios2 = jest.spyOn(axios, "get");
mockAxios2.mockImplementationOnce(() =>
  Promise.resolve({})
);

const store = global.store

const data = {
  'Model.1': {
    1: {title: 'Entry 1', color: 'red'},
    2: {title: 'Entry 2', color: 'blue'},
    3: {title: 'Entry 3', color: 'blue'},
    4: {title: 'Entry 4', color: 'blue'},
    5: {title: 'Entry 5', color: 'blue'},
    6: {title: 'Entry 6', color: 'blue'},
    7: {title: 'Entry 7', color: 'blue'},
  }
}

const getEntry = (model, pk) => {
  return data[model][pk.id];
}

const new_entries = {}
const NEW_DATA = [
  {title: 'Entry 8', color: 'red'},
  {title: 'Entry 9', color: 'red'},
  {title: 'Entry 10', color: 'red'},
  {title: 'Entry 11', color: 'red'},
  {title: 'Entry 12', color: 'red'},
  {title: 'Entry 13', color: 'red'},
  {title: 'Entry 14', color: 'red'},
  {title: 'Entry 15', color: 'red'},
  {title: 'Entry 16', color: 'red'},
];
const getNewEntries = (model) => {
  return new_entries[model];
}

describe('furet-ui-resource-list component', () => {

  store.commit('UPDATE_RESOURCES', {'definitions': [{
    'id': 1, 
    'type': 'list', 
    'title': 'Templates', 
    'model': 'Model.1', 
    'filters': [], 
    'tags': [], 
    'headers': [
      {
        'hidden': false, 
        'name': 'title', 
        'label': 'Title', 
        'component': 'furet-ui-field', 
        'type': 'string', 
        'numeric': false, 
        'tooltip': null, 
        'sortable': true
      },
      {
        'hidden': false, 
        'name': 'color', 
        'label': 'Color', 
        'component': 'furet-ui-field', 
        'type': 'string', 
        'numeric': false, 
        'tooltip': null, 
        'sortable': true
      }
    ], 
    'fields': ['id', 'title', 'color']
  }]});

  store.commit('UPDATE_RESOURCES', {'definitions': [{
    'id': 2, 
    'type': 'list', 
    'title': 'Templates', 
    'model': 'Model.1', 
    'filters': [], 
    'tags': [], 
    'buttons': [
      {
        icon: 'th',
        label: 'Test',
        call: 'test',
      },
    ],
    'headers': [
      {
        'hidden': false, 
        'name': 'title', 
        'label': 'Title', 
        'component': 'furet-ui-field', 
        'type': 'string', 
        'numeric': false, 
        'tooltip': null, 
        'sortable': true
      },
    ], 
    'fields': ['id', 'title', 'color']
  }]});

  store.commit('UPDATE_RESOURCES', {'definitions': [{
    'id': 3, 
    'type': 'list', 
    'title': 'Templates', 
    'model': 'Model.1', 
    'filters': [], 
    'tags': [], 
    'buttons': [
      {
        icon: 'th',
        label: 'Test',
        'open-resource': 'test',
      },
    ],
    'headers': [
      {
        'hidden': false, 
        'name': 'title', 
        'label': 'Title', 
        'component': 'furet-ui-field', 
        'type': 'string', 
        'numeric': false, 
        'tooltip': null, 
        'sortable': true
      },
    ], 
    'fields': ['id', 'title', 'color']
  }]});

  const getWrapper = (resourceId) => {
    const id = resourceId || 1;
    return mount(getComponentPrototype("furet-ui-resource-list"), {
      store,
      localVue,
      router,
      i18n,
      propsData: {
        id,
        manager: {},
      },
      provide: {
        getEntry,
        getNewEntries
      }
    });
  }


  beforeEach(() => {
    mockAxios.mockClear();
    mockAxios2.mockClear();
    new_entries["Model.1"] = [];
  });

  it('furet-ui-resource-list snapshot without data', () => {
      const wrapper = getWrapper();
      expect(wrapper.element).toMatchSnapshot();
  });
  it('furet-ui-resource-list snapshot without data with button', () => {
      const wrapper = getWrapper(2);
      expect(wrapper.element).toMatchSnapshot();
  });
  it('furet-ui-resource-list snapshot with data', () => {
    const response = {
      pks: [{id: 1}, {id: 2}],
      total: 2,
      data: [] // useless for test because getEntry is mocked
    };

    const wrapper = getWrapper();
    const list = wrapper.vm.$refs.list;
    wrapper.vm.api_formater(list, response)
    wrapper.vm.$nextTick(() => {
      expect(wrapper.element).toMatchSnapshot();
    })
  });
  it('furet-ui-resource-list.api_formater without data', () => {
      const wrapper = getWrapper();
      const obj = {
      };
      wrapper.vm.api_formater(obj, {});
      expect(obj.total).toBe(0);
      expect(obj.data.length).toBe(0);
      expect(obj.number_created).toBe(0);
      expect(obj.number_updated).toBe(0);
      expect(obj.number_deleted).toBe(0);
      expect(obj.number_linked).toBe(0);
      expect(obj.number_unlinked).toBe(0);
  });
  it('furet-ui-resource-list.api_formater exist object', () => {
      const wrapper = getWrapper();
      const obj = {
        data : 1,
        number_created : 1,
        number_updated : 1,
        number_deleted : 1,
        number_linked : 1,
        number_unlinked : 1,
      };
      wrapper.vm.api_formater(obj, {});
      expect(obj.total).toBe(0);
      expect(obj.data.length).toBe(0);
      expect(obj.number_created).toBe(0);
      expect(obj.number_updated).toBe(0);
      expect(obj.number_deleted).toBe(0);
      expect(obj.number_linked).toBe(0);
      expect(obj.number_unlinked).toBe(0);
  });

  it('furet-ui-resource-list.api_formater page 1 with data and new entries', () => {
    const wrapper = getWrapper();
    const obj = {
      page: 1,
      perPage: 5,
    }
    new_entries["Model.1"] = NEW_DATA
    wrapper.vm.api_formater(obj, {total: 7, pks: [{id: 1}, {id: 2}, {id: 3}, {id: 4}, {id: 5}]});
    expect(obj.total).toBe(16);
    expect(obj.data.length).toBe(5);
    expect(
      obj.data.reduce(function(accumulator, currentValue) {
        accumulator.push(currentValue.title);
        return accumulator;
      }, [])
    ).toEqual(
      ["Entry 1", "Entry 2", "Entry 3", "Entry 4", "Entry 5"]
    );
  });
  it('furet-ui-resource-list.api_formater page 2 with data and new entries', () => {
    const wrapper = getWrapper();
    const obj = {
      page: 2,
      perPage: 5,
    }
    new_entries["Model.1"] = NEW_DATA
    wrapper.vm.api_formater(obj, {total: 7, pks: [{id: 6}, {id: 7}]});
    expect(obj.total).toBe(16);
    expect(obj.data.length).toBe(5);
    expect(
      obj.data.reduce(function(accumulator, currentValue) {
        accumulator.push(currentValue.title);
        return accumulator;
      }, [])
    ).toEqual(
      ["Entry 6", "Entry 7", "Entry 8", "Entry 9", "Entry 10"]
    );
  });
  it('furet-ui-resource-list.api_formater page 3 with data and new entries', () => {
    const wrapper = getWrapper();
    const obj = {
      page: 3,
      perPage: 5,
    }
    new_entries["Model.1"] = NEW_DATA
    wrapper.vm.api_formater(obj, {total: 7, pks: []});
    expect(obj.total).toBe(16);
    expect(obj.data.length).toBe(5);
    expect(
      obj.data.reduce(function(accumulator, currentValue) {
        accumulator.push(currentValue.title);
        return accumulator;
      }, [])
    ).toEqual(
      ["Entry 11", "Entry 12", "Entry 13", "Entry 14", "Entry 15"]
    );
  });
  it('furet-ui-resource-list.api_formater page 4 with data and new entries', () => {
    const wrapper = getWrapper();
    const obj = {
      page: 4,
      perPage: 5,
    }
    new_entries["Model.1"] = NEW_DATA
    wrapper.vm.api_formater(obj, {total: 7, pks: []});
    expect(obj.total).toBe(16);
    expect(obj.data.length).toBe(1);
    expect(
      obj.data.reduce(function(accumulator, currentValue) {
        accumulator.push(currentValue.title);
        return accumulator;
      }, [])
    ).toEqual(
      ["Entry 16"]
    );
  });
  it('furet-ui-resource-list.api_formater page 1 with no data and new entries', () => {
    const wrapper = getWrapper();
    const obj = {
      page: 1,
      perPage: 5,
    }
    new_entries["Model.1"] = NEW_DATA
    wrapper.vm.api_formater(obj, {total: 0, pks: []});
    expect(obj.total).toBe(9);
    expect(obj.data.length).toBe(5);
    expect(
      obj.data.reduce(function(accumulator, currentValue) {
        accumulator.push(currentValue.title);
        return accumulator;
      }, [])
    ).toEqual(
      ["Entry 8", "Entry 9", "Entry 10", "Entry 11", "Entry 12", ]
    );
  });
  it('furet-ui-resource-list.api_formater page 2 with no data and new entries', () => {
    const wrapper = getWrapper();
    const obj = {
      page: 2,
      perPage: 5,
    }
    new_entries["Model.1"] = NEW_DATA
    wrapper.vm.api_formater(obj, {total: 0, pks: []});
    expect(obj.total).toBe(9);
    expect(obj.data.length).toBe(4);
    expect(
      obj.data.reduce(function(accumulator, currentValue) {
        accumulator.push(currentValue.title);
        return accumulator;
      }, [])
    ).toEqual(
      ["Entry 13", "Entry 14", "Entry 15", "Entry 16" ]
    );
  });
  it('furet-ui-resource-list breadscrumb', () => {
    const wrapper = getWrapper();
    expect(wrapper.vm.getBreadcrumbInfo().icon).toBe('list');
  });
  it('furet-ui-resource-list refresh', () => {
    const wrapper = getWrapper();
    wrapper.vm.refresh()
    expect(mockAxios2).toBeCalledWith(
      "/furet-ui/resource/1/crud", 
      {
        params: {
          'context[fields]': "id,title,color",
          'context[model]': "Model.1",
          limit: 25,
          offset: 0,
        }
      }
    );
  });
  it('furet-ui-resource-list toggleHiddenColumn', () => {
    const wrapper = getWrapper();
    wrapper.vm.toggleHiddenColumn({})
  });
  it('furet-ui-resource-list click button call', async () => {
    const response = {
      pks: [{id: 1}],
      total: 1,
      data: [] // useless for test because getEntry is mocked
    };

    const wrapper = getWrapper(2);
    const list = wrapper.vm.$refs.list;
    wrapper.vm.api_formater(list, response)
    await wrapper.vm.$nextTick()
    const button = wrapper.find('a.furet-ui-list-button');
    button.trigger('click')
    expect(mockAxios).toBeCalledWith("/furet-ui/resource/2/model/Model.1/call/test", {});
  });
  it('furet-ui-resource-list click button open resource', async () => {
    const response = {
      pks: [{id: 1}],
      total: 1,
      data: [] // useless for test because getEntry is mocked
    };

    const wrapper = getWrapper(3);
    const list = wrapper.vm.$refs.list;
    wrapper.vm.api_formater(list, response)
    await wrapper.vm.$nextTick()
    const button = wrapper.find('a.furet-ui-list-button');
    button.trigger('click')
    expect(mockAxios).toBeCalledWith(
      "/furet-ui/open/resource/test", 
      {
        params: {},
        resource_type: undefined,
        route: null,
      }
    );
  });
});
