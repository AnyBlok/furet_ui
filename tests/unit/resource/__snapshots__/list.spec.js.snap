// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`furet-ui-resource-list component furet-ui-resource-list snapshot with data 1`] = `
<section
  id="furet-ui-list"
>
  <div
    class="container"
  >
    <!---->
  </div>
   
  <header
    id="furet-ui-page-multi-entries-header"
  >
    <div
      class="level"
    >
      <div
        class="level-left"
      >
        <h2
          class="level-item is-size-3"
        >
          Templates
        </h2>
         
        <span
          class="level-item"
        >
           
        </span>
      </div>
       
      <div
        class="level-right"
      >
        <div
          class="level-item"
        >
          <div
            class="field is-grouped-multiline is-grouped"
          >
            <!---->
              
            <!---->
          </div>
        </div>
         
        <!---->
         
        <div
          class="level-item buttons"
        >
          <a
            class="button"
            id="furet-ui-page-multi-entries-header-refresh"
          >
            <span
              class="icon"
            >
              <i
                class="fa fa-redo fa-lg"
              />
            </span>
          </a>
           
          <!---->
        </div>
      </div>
    </div>
     
    <div
      class="collapse panel is-pulled-right"
    >
      <div
        class="collapse-trigger"
      />
      <div
        class="collapse-content"
        id=""
        style="display: none;"
      >
        <div
          class="tags"
        />
      </div>
    </div>
     
    <div
      class="buttons is-grouped is-centered"
    >
      <!---->
       
    </div>
  </header>
    
  <div
    class="b-table"
  >
    <div
      class="field table-mobile-sort"
    >
      <div
        class="field has-addons"
      >
        <div
          class="control is-expanded"
        >
          <span
            class="select is-fullwidth"
          >
            <select>
              <!---->
               
              <!---->
              <option
                value="[object Object]"
              >
                
                    Title
                    
                <!---->
              </option>
              <option
                value="[object Object]"
              >
                
                    Color
                    
                <!---->
              </option>
            </select>
          </span>
           
          <!---->
        </div>
         
        <div
          class="control"
        >
          <!---->
        </div>
      </div>
    </div>
     
    <div
      class="top level"
    >
      <div
        class="level-left"
      >
        <div
          class="field is-grouped-multiline is-grouped"
        >
          <!---->
           
          <div
            class="control"
          >
            <strong>
              [object Promise] :
            </strong>
            
        2
      
          </div>
           
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
        </div>
      </div>
       
      <div
        class="level-right"
      >
        <div
          class="level-item"
        >
          <nav
            class="pagination"
          >
            <a
              class="pagination-link pagination-previous"
              disabled="disabled"
              href="#"
              role="button"
            >
              <span
                aria-hidden="true"
                class="icon"
              >
                <i
                  class="fa fa-angle-left fa-lg"
                />
              </span>
            </a>
             
            <a
              class="pagination-link pagination-next"
              disabled="disabled"
              href="#"
              role="button"
            >
              <span
                aria-hidden="true"
                class="icon"
              >
                <i
                  class="fa fa-angle-right fa-lg"
                />
              </span>
            </a>
             
            <ul
              class="pagination-list"
            >
              <!---->
               
              <!---->
               
              <li>
                <a
                  aria-current="true"
                  class="pagination-link is-current "
                  href="#"
                  role="button"
                >
                  1
                </a>
              </li>
               
              <!---->
               
              <!---->
            </ul>
          </nav>
        </div>
      </div>
    </div>
     
    <div
      class="table-wrapper has-mobile-cards"
    >
      <table
        class="table is-striped is-hoverable"
      >
        <thead>
          <tr>
            <!---->
             
            <!---->
             
            <th
              class=""
            >
              <div
                class="th-wrap"
              >
                 
                <!---->
              </div>
            </th>
            <th
              class="is-sortable"
            >
              <div
                class="th-wrap"
              >
                Title 
                <!---->
              </div>
            </th>
            <th
              class="is-sortable"
            >
              <div
                class="th-wrap"
              >
                Color 
                <!---->
              </div>
            </th>
             
            <!---->
          </tr>
           
          <!---->
           
          <!---->
        </thead>
         
        <tbody>
          <tr
            class="is-unmodified"
            draggable="false"
          >
            <!---->
             
            <!---->
             
            <td
              class="is-action"
            >
              <a
                class="button is-outlined"
              >
                <span
                  class="icon is-small"
                >
                  <i
                    class="fa fa-redo-alt"
                  />
                </span>
              </a>
            </td>
             
            <td
              class="is-list-cell"
              data-label="Title"
            >
              <div>
                <span>
                  Entry 1
                </span>
              </div>
            </td>
            <td
              class="is-list-cell"
              data-label="Color"
            >
              <div>
                <span>
                  red
                </span>
              </div>
            </td>
             
            <!---->
          </tr>
           
          <!---->
           
          <!---->
          <tr
            class="is-unmodified"
            draggable="false"
          >
            <!---->
             
            <!---->
             
            <td
              class="is-action"
            >
              <a
                class="button is-outlined"
              >
                <span
                  class="icon is-small"
                >
                  <i
                    class="fa fa-redo-alt"
                  />
                </span>
              </a>
            </td>
             
            <td
              class="is-list-cell"
              data-label="Title"
            >
              <div>
                <span>
                  Entry 2
                </span>
              </div>
            </td>
            <td
              class="is-list-cell"
              data-label="Color"
            >
              <div>
                <span>
                  blue
                </span>
              </div>
            </td>
             
            <!---->
          </tr>
           
          <!---->
           
          <!---->
        </tbody>
         
        <!---->
      </table>
    </div>
     
    <div
      class="level"
    >
      <div
        class="level-left"
      >
        <div
          class="field is-grouped-multiline is-grouped"
        >
          <!---->
           
          <div
            class="control"
          >
            <strong>
              [object Promise] :
            </strong>
            
        2
      
          </div>
           
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
        </div>
      </div>
       
      <div
        class="level-right"
      >
        <div
          class="level-item"
        >
          <nav
            class="pagination"
          >
            <a
              class="pagination-link pagination-previous"
              disabled="disabled"
              href="#"
              role="button"
            >
              <span
                aria-hidden="true"
                class="icon"
              >
                <i
                  class="fa fa-angle-left fa-lg"
                />
              </span>
            </a>
             
            <a
              class="pagination-link pagination-next"
              disabled="disabled"
              href="#"
              role="button"
            >
              <span
                aria-hidden="true"
                class="icon"
              >
                <i
                  class="fa fa-angle-right fa-lg"
                />
              </span>
            </a>
             
            <ul
              class="pagination-list"
            >
              <!---->
               
              <!---->
               
              <li>
                <a
                  aria-current="true"
                  class="pagination-link is-current "
                  href="#"
                  role="button"
                >
                  1
                </a>
              </li>
               
              <!---->
               
              <!---->
            </ul>
          </nav>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`furet-ui-resource-list component furet-ui-resource-list snapshot without data 1`] = `
<section
  id="furet-ui-list"
>
  <div
    class="container"
  >
    <!---->
  </div>
   
  <header
    id="furet-ui-page-multi-entries-header"
  >
    <div
      class="level"
    >
      <div
        class="level-left"
      >
        <h2
          class="level-item is-size-3"
        >
          Templates
        </h2>
         
        <span
          class="level-item"
        >
           
        </span>
      </div>
       
      <div
        class="level-right"
      >
        <div
          class="level-item"
        >
          <div
            class="field is-grouped-multiline is-grouped"
          >
            <!---->
              
            <!---->
          </div>
        </div>
         
        <!---->
         
        <div
          class="level-item buttons"
        >
          <a
            class="button"
            id="furet-ui-page-multi-entries-header-refresh"
          >
            <span
              class="icon"
            >
              <i
                class="fa fa-redo fa-lg"
              />
            </span>
          </a>
           
          <!---->
        </div>
      </div>
    </div>
     
    <div
      class="collapse panel is-pulled-right"
    >
      <div
        class="collapse-trigger"
      />
      <div
        class="collapse-content"
        id=""
        style="display: none;"
      >
        <div
          class="tags"
        />
      </div>
    </div>
     
    <div
      class="buttons is-grouped is-centered"
    >
      <!---->
       
    </div>
  </header>
    
  <div
    class="b-table"
  >
    <!---->
     
    <div
      class="top level"
    >
      <div
        class="level-left"
      >
        <div
          class="field is-grouped-multiline is-grouped"
        >
          <!---->
           
          <div
            class="control"
          >
            <strong>
              [object Promise] :
            </strong>
            
        0
      
          </div>
           
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
        </div>
      </div>
       
      <div
        class="level-right"
      >
        <div
          class="level-item"
        >
          <nav
            class="pagination"
          >
            <a
              class="pagination-link pagination-previous"
              disabled="disabled"
              href="#"
              role="button"
            >
              <span
                aria-hidden="true"
                class="icon"
              >
                <i
                  class="fa fa-angle-left fa-lg"
                />
              </span>
            </a>
             
            <a
              class="pagination-link pagination-next"
              disabled="disabled"
              href="#"
              role="button"
            >
              <span
                aria-hidden="true"
                class="icon"
              >
                <i
                  class="fa fa-angle-right fa-lg"
                />
              </span>
            </a>
             
            <ul
              class="pagination-list"
            >
              <!---->
               
              <!---->
                
              <!---->
               
              <!---->
            </ul>
          </nav>
        </div>
      </div>
    </div>
     
    <div
      class="table-wrapper has-mobile-cards"
    >
      <table
        class="table is-striped"
      >
        <!---->
         
        <tbody>
          <tr
            class="is-empty"
          >
            <td
              colspan="0"
            >
              <section
                class="section"
              >
                <div
                  class="content has-text-grey has-text-centered"
                >
                  <p>
                    [object Promise]
                  </p>
                </div>
              </section>
            </td>
          </tr>
        </tbody>
         
        <!---->
      </table>
    </div>
     
    <div
      class="level"
    >
      <div
        class="level-left"
      >
        <div
          class="field is-grouped-multiline is-grouped"
        >
          <!---->
           
          <div
            class="control"
          >
            <strong>
              [object Promise] :
            </strong>
            
        0
      
          </div>
           
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
        </div>
      </div>
       
      <div
        class="level-right"
      >
        <div
          class="level-item"
        >
          <nav
            class="pagination"
          >
            <a
              class="pagination-link pagination-previous"
              disabled="disabled"
              href="#"
              role="button"
            >
              <span
                aria-hidden="true"
                class="icon"
              >
                <i
                  class="fa fa-angle-left fa-lg"
                />
              </span>
            </a>
             
            <a
              class="pagination-link pagination-next"
              disabled="disabled"
              href="#"
              role="button"
            >
              <span
                aria-hidden="true"
                class="icon"
              >
                <i
                  class="fa fa-angle-right fa-lg"
                />
              </span>
            </a>
             
            <ul
              class="pagination-list"
            >
              <!---->
               
              <!---->
                
              <!---->
               
              <!---->
            </ul>
          </nav>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`furet-ui-resource-list component furet-ui-resource-list snapshot without data with button 1`] = `
<section
  id="furet-ui-list"
>
  <div
    class="container"
  >
    <!---->
  </div>
   
  <header
    id="furet-ui-page-multi-entries-header"
  >
    <div
      class="level"
    >
      <div
        class="level-left"
      >
        <h2
          class="level-item is-size-3"
        >
          Templates
        </h2>
         
        <span
          class="level-item"
        >
           
        </span>
      </div>
       
      <div
        class="level-right"
      >
        <div
          class="level-item"
        >
          <div
            class="field is-grouped-multiline is-grouped"
          >
            <!---->
              
            <!---->
          </div>
        </div>
         
        <!---->
         
        <div
          class="level-item buttons"
        >
          <a
            class="button"
            id="furet-ui-page-multi-entries-header-refresh"
          >
            <span
              class="icon"
            >
              <i
                class="fa fa-redo fa-lg"
              />
            </span>
          </a>
           
          <!---->
        </div>
      </div>
    </div>
     
    <div
      class="collapse panel is-pulled-right"
    >
      <div
        class="collapse-trigger"
      />
      <div
        class="collapse-content"
        id=""
        style="display: none;"
      >
        <div
          class="tags"
        />
      </div>
    </div>
     
    <div
      class="buttons is-grouped is-centered"
    >
      <!---->
       
      <a
        class="furet-ui-list-button button is-primary is-outlined"
      >
        <span
          class="icon"
        >
          <span
            class="icon"
          >
            <i
              class="fa fa-th fa-lg"
            />
          </span>
        </span>
         
        <span>
          Test
        </span>
      </a>
    </div>
  </header>
    
  <div
    class="b-table"
  >
    <!---->
     
    <div
      class="top level"
    >
      <div
        class="level-left"
      >
        <div
          class="field is-grouped-multiline is-grouped"
        >
          <!---->
           
          <div
            class="control"
          >
            <strong>
              [object Promise] :
            </strong>
            
        0
      
          </div>
           
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
        </div>
      </div>
       
      <div
        class="level-right"
      >
        <div
          class="level-item"
        >
          <nav
            class="pagination"
          >
            <a
              class="pagination-link pagination-previous"
              disabled="disabled"
              href="#"
              role="button"
            >
              <span
                aria-hidden="true"
                class="icon"
              >
                <i
                  class="fa fa-angle-left fa-lg"
                />
              </span>
            </a>
             
            <a
              class="pagination-link pagination-next"
              disabled="disabled"
              href="#"
              role="button"
            >
              <span
                aria-hidden="true"
                class="icon"
              >
                <i
                  class="fa fa-angle-right fa-lg"
                />
              </span>
            </a>
             
            <ul
              class="pagination-list"
            >
              <!---->
               
              <!---->
                
              <!---->
               
              <!---->
            </ul>
          </nav>
        </div>
      </div>
    </div>
     
    <div
      class="table-wrapper has-mobile-cards"
    >
      <table
        class="table is-striped"
      >
        <!---->
         
        <tbody>
          <tr
            class="is-empty"
          >
            <td
              colspan="0"
            >
              <section
                class="section"
              >
                <div
                  class="content has-text-grey has-text-centered"
                >
                  <p>
                    [object Promise]
                  </p>
                </div>
              </section>
            </td>
          </tr>
        </tbody>
         
        <!---->
      </table>
    </div>
     
    <div
      class="level"
    >
      <div
        class="level-left"
      >
        <div
          class="field is-grouped-multiline is-grouped"
        >
          <!---->
           
          <div
            class="control"
          >
            <strong>
              [object Promise] :
            </strong>
            
        0
      
          </div>
           
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
        </div>
      </div>
       
      <div
        class="level-right"
      >
        <div
          class="level-item"
        >
          <nav
            class="pagination"
          >
            <a
              class="pagination-link pagination-previous"
              disabled="disabled"
              href="#"
              role="button"
            >
              <span
                aria-hidden="true"
                class="icon"
              >
                <i
                  class="fa fa-angle-left fa-lg"
                />
              </span>
            </a>
             
            <a
              class="pagination-link pagination-next"
              disabled="disabled"
              href="#"
              role="button"
            >
              <span
                aria-hidden="true"
                class="icon"
              >
                <i
                  class="fa fa-angle-right fa-lg"
                />
              </span>
            </a>
             
            <ul
              class="pagination-list"
            >
              <!---->
               
              <!---->
                
              <!---->
               
              <!---->
            </ul>
          </nav>
        </div>
      </div>
    </div>
  </div>
</section>
`;
